### PR TITLE
Notify Slack of failures via pwsh Invoke-RestMethod

### PR DESCRIPTION
**Pull request by DavidBoike**

### CODE REVIEW

File: .github/workflows/build-samples-and-snippets.yml [hunk 1]
```diff
@@ -22,17 +22,22 @@ jobs:
...
+          $body = @{
+            channel = 'docs'
+            text = "Failed to build samples and snippets on master: https://github.com/Particular/docs.particular.net/actions/workflows/build-samples-and-snippets.yml?query=branch%3Amaster"
+            username = 'Check Samples & Snippets'
+            icon_emoji = 'github_actions'
+            unfurl_links = false
+            unfurl_media = false
+          } | ConvertTo-Json
+          $result = Invoke-RestMethod -Method POST -Uri https://slack.com/api/chat.postMessage -ContentType "application/json; charset=utf-8" -Headers $headers -Body $body
+          Write-Output $result
```
Comment (danielmarbach) [inline .github/workflows/build-samples-and-snippets.yml:39]: Could we ever leak tokens here into the output? Other than that it looks good to me

Comment (adamralph) [reply .github/workflows/build-samples-and-snippets.yml:39]: AFAIK GitHub Actions masks all secrets from the logs.

File: .github/workflows/build-samples-and-snippets.yml [hunk 2]
```diff
@@ -22,17 +22,22 @@ jobs:
...
-          SLACK_WEBHOOK_URL: ${{ secrets.SLACK_DOCS_CHANNEL_WEBHOOK }}
+        shell: pwsh
+        run: |
+          $headers = @{
+              'Authorization' = "Bearer ${{ secrets.GITHUB_TOKEN }}"
+          }
+          $body = @{
+            channel = 'docs'
+            text = "Failed to build samples and snippets on master: https://github.com/Particular/docs.particular.net/actions/workflows/build-samples-and-snippets.yml?query=branch%3Amaster"
+            username = 'Check Samples & Snippets'
```
Comment (adamralph) [inline .github/workflows/build-samples-and-snippets.yml:33]:
```suggestion
            username = 'Build samples and snippets'
```

File: .github/workflows/verify-master.yml [hunk 3]
```diff
@@ -23,17 +23,22 @@ jobs:
...
-          SLACK_WEBHOOK_URL: ${{ secrets.SLACK_DOCS_CHANNEL_WEBHOOK }}
+        shell: pwsh
+        run: |
+          $headers = @{
+              'Authorization' = "Bearer ${{ secrets.GITHUB_TOKEN }}"
+          }
+          $body = @{
+            channel = 'docs'
+            text = "Content verification has failed on master: https://github.com/Particular/docs.particular.net/actions/workflows/verify-master.yml?query=branch%3Amaster"
+            username = 'Content Verification'
```
Comment (adamralph) [inline .github/workflows/verify-master.yml:34]:
```suggestion
            username = 'Content verification'
```

File: .github/workflows/verify-master.yml [hunk 4]
```diff
@@ -48,13 +53,20 @@ jobs:
...
-          SLACK_WEBHOOK_URL: ${{ secrets.SLACK_DOCS_CHANNEL_WEBHOOK }}
+        shell: pwsh
+        run: |
+          $headers = @{
+              'Authorization' = "Bearer ${{ secrets.GITHUB_TOKEN }}"
+          }
+          $body = @{
+            channel = 'docs'
+            text = "Integrity Tests have failed on master: https://github.com/Particular/docs.particular.net/actions/workflows/verify-master.yml?query=branch%3Amaster"
+            username = 'Integrity Tests'
```
Comment (adamralph) [inline .github/workflows/verify-master.yml:64]:
```suggestion
            username = 'Integrity tests'
```

File: .github/workflows/verify-master.yml [hunk 5]
```diff
@@ -48,13 +53,20 @@ jobs:
...
-          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
-          SLACK_WEBHOOK_URL: ${{ secrets.SLACK_DOCS_CHANNEL_WEBHOOK }}
+        shell: pwsh
+        run: |
+          $headers = @{
+              'Authorization' = "Bearer ${{ secrets.GITHUB_TOKEN }}"
+          }
+          $body = @{
+            channel = 'docs'
+            text = "Integrity Tests have failed on master: https://github.com/Particular/docs.particular.net/actions/workflows/verify-master.yml?query=branch%3Amaster"
```
Comment (adamralph) [inline .github/workflows/verify-master.yml:63]:
```suggestion
            text = "Integrity tests have failed on master: https://github.com/Particular/docs.particular.net/actions/workflows/verify-master.yml?query=branch%3Amaster"
```

File: .github/workflows/build-samples-and-snippets.yml (hunk 1 above)
Comment (danielmarbach) [reply .github/workflows/build-samples-and-snippets.yml:39]: Well, it depends. Run into huge issues when using multiple jobs and passing of secrets over outputs. But yes, most of the time "it works"

**Pull request merged**